main.tf

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+resource "google_dns_managed_zone" "dns_managed_zone" {
+  count       = "${var.enable_dns_managed_zone && length(var.dns_name) >0 && length(var.description) >0 ? 1 : 0}"
```

Comment:
Since this is a new module, let's nip this anti-pattern in the bud. Complex conditionals based on optional variables should be replaced with modules which the calling configuration can optionally include.